docs/motivation.md

```diff
@@ -0,0 +1,535 @@
+# The Verona language: Features and Motivation 
+
+Verona is an object oriented reference capability language with structural
+subtyping and bounded polymorphism. Here we summarize each part and try to
+motivate the design decisions taken so far. We also pose design questions that
+are yet to be answered.
+
+## Structural type system
+
+We use a structural and algebraic type system rather than a nominal one, since
+we find it more flexible.
+
+### Traits
+Verona is a structural trait based type system, i.e. types can be described
+using combination of methods on an object. E.g. we can describe a type
+implementing a `toString` method as
+```verona
+type ToString = {
+    toString(self: Self): String
+}
+```
+and a typed describing numbers implementing addition and subtraction with
+```verona
+type Number = {
+    add(self: Self, other: Number): Number
+    sub(self: Self, other: Number): Number
+}
+```
+Traits also support default implementations. E.g. we can add a default
+implementation of multiplication as
+
+* TODO finish paragraph
+
+
+### Conjunction types
+We can easily combine traits using conjunctions, so if we e.g. want to describe
+something implementing both `ToString` and `Number`, this can be written as
+```verona
+type ToStringNumber = ToString & Number
+```
+
+This gives us a flexible way to define combinations types on the fly, with an
+intuitive syntax. E.g. if we want to describe the type signature of a function
+`f` printing two numbers and then returning the result of adding them
+together, this would be
+```verona
+f(n1: Number & ToString, n2: Number & ToString) : Number
+// TODO add ToStringNumber version
+```
+
+In contrast, in a nominal system like java, we would have to define a new
+interface `ToStringNumber` that extends both `Number` and `ToString`:
+```java
+interface ToStringNumber extends Number, ToString { ... }
+
+class C {
+    static Number f(ToStringNumber n1, ToStringNumber n2) { ... }
+}
+```
+
+In Verona, a trait with multiple methods can be further simplified by breaking
+up each method into what we call an "atomic" trait and combining them with a
+conjunction. E.g. the definition of `Number` above is simply syntactic sugar for
+```verona
+type Number = {
+    add(self: Self, other: Number): Number
+} & {
+    sub(self: Self, other: Number): Number
+}
+```
```

Comment:
It probably doesn't make much difference to most people. On the other hand, I don't think it hurts to mention it?

Reply:
I think it might even help with the intuition for `&`, reading it again now. But maybe it should be more of a note than a proposed simplification (I would argue that it is less simple to compose atomic traits than to write a composite trait).

Reply:
yeah, i agree, i'll see if i have time to write something before the meeting

Reply:
"syntactic sugar" ==> "can also be understood as" 

I think our readers do not care about syntactic sugar.